packages/components/src/utils/OSfunctions.test.tsx

```diff
@@ -0,0 +1,294 @@
+          longitude: location.longitude,
+        })
+
+        expect(url).toBe(
+          'https://maps.apple.com/?t=m&daddr=%2BTibor+Rubin+VA+Medical+Center%2B33.7764681%2C-118.1189664',
+        )
+      })
+
+      it('forms URL with all props', () => {
+        url = FormDirectionsUrl(location)
+
+        expect(url).toBe(
+          'https://maps.apple.com/?t=m&daddr=5901+E+7th+St+Long+Beach+CA+90822%2BTibor+Rubin+VA+Medical+Center%2B33.7764681%2C-118.1189664',
+        )
+      })
+    })
+
+    describe('Google Maps tests', () => {
+      it('forms URL with just name/address', () => {
+        Platform.OS = 'android'
+        url = FormDirectionsUrl({
+          name: location.name,
+          address: location.address,
+        })
+
+        expect(url).toBe(
+          'https://www.google.com/maps/dir/?api=1&destination=5901+E+7th+St+Long+Beach+CA+90822',
+        )
+      })
+
+      it('forms URL with just name/latitude/longitude', () => {
+        url = FormDirectionsUrl({
+          name: location.name,
+          latitude: location.latitude,
+          longitude: location.longitude,
+        })
+
+        expect(url).toBe(
+          'https://www.google.com/maps/dir/?api=1&destination=33.7764681%2C-118.1189664',
+        )
+      })
+
+      it('forms URL with all props', () => {
+        url = FormDirectionsUrl(location)
+
+        expect(url).toBe(
+          'https://www.google.com/maps/dir/?api=1&destination=5901+E+7th+St+Long+Beach+CA+90822',
+        )
+      })
+    })
+  })
+
+  type UseExternalLinkButtonProps = {
+    url: string
+    analytics?: LinkAnalytics
+    text?: leaveAppPromptText
+  }
+
+  /** Dummy component modeled off Link to access useExternalLink hook logic */
+  const UseExternalLinkButton: FC<UseExternalLinkButtonProps> = ({
+    url,
+    analytics,
+    text,
+  }) => {
+    const launchExternalLink = useExternalLink()
+    const onPress = async (): Promise<void> => {
+      launchExternalLink(url, analytics, text)
+    }
+    return (
+      <ComponentWrapper>
+        <Pressable testID="useExternalLinkButton" onPress={onPress}>
+          <Text>useExternalLinkButton</Text>
+        </Pressable>
+      </ComponentWrapper>
+    )
+  }
+
+  describe('useExternalLink tests', () => {
+    const onPressAnalyticMock = jest.fn()
+    const onCancelAnalyticMock = jest.fn()
+    const onConfirmAnalyticMock = jest.fn()
+
+    const commonProps: UseExternalLinkButtonProps = {
+      url: 'https://www.va.com',
+      analytics: {
+        onPress: onPressAnalyticMock,
+        onCancel: onCancelAnalyticMock,
+        onConfirm: onConfirmAnalyticMock,
+      },
+      text: {
+        title: 'Title override',
+        body: 'Body override',
+        cancel: 'Cancel override',
+        confirm: 'Confirm override',
+      },
+    }
+
+    afterEach(() => {
+      jest.resetAllMocks()
+    })
+
+    it('should call RN Linking with no RN Alert prompt on non-URL variety', () => {
```

Comment:
Should we rename this "tel variety" or "non-HTTP variety"? Test description is a little confusing to see "non-URL" then see the url prop used.

Reply:
Updated to `non-HTTP`.